aptos-move/framework/aptos-framework/sources/SharedAccount.move

```diff
@@ -0,0 +1,153 @@
+module AptosFramework::SharedAccount {
+    use Std::Errors;
+    use Std::Signer;
+    use Std::Vector;
+    use AptosFramework::Account;
+    use AptosFramework::Coin;
+    use AptosFramework::SimpleMap::{Self, SimpleMap};
+    use AptosFramework::ResourceAccount;
+
+    struct Shares has store {
+        numerator: u64,
+        denominator: u64,
+    }
+
+    // Resource representing a shared account
+    struct SharedAccount has key {
+        share_record: SimpleMap<address, Shares>,
+    }
+
+    const EINVALID_INIT_INPUT: u64 = 0;
+    const EACCOUNT_DNE: u64 = 1;
+    const EADDRESS_ALREADY_EXISTS: u64 = 2;
+    const EINVALID_NUMERATOR_DENOMINATOR_COMBINATIONS: u64 = 3;
+    const EINVALID_SIGNER: u64 = 4;
+    const EINSUFFICIENT_BALANCE: u64 = 5;
+
+    public(script) fun initialize_shared_account(source: &signer, seed: vector<u8>): (signer) {
+        assert!(Account::exists_at(Signer::address_of(source)), Errors::invalid_argument(EACCOUNT_DNE));
+        let resource_signer = ResourceAccount::create_resource_account(source, seed, Vector::empty());
+        if (!exists<SharedAccount>(Signer::address_of(&resource_signer))) {
+            move_to(
+                &resource_signer,
+                SharedAccount {
+                    share_record: SimpleMap::create(),
+                }
+            )
+        };
+        resource_signer
+    }
+
+    public(script) fun initialize(source: &signer, seed: vector<u8>, addresses: vector<address>, numerators: vector<u64>, denominator: u64): (signer) acquires SharedAccount {
```

Comment:
Ah, I see.  Let's call `numerators` -> `shares`, and call `denominator` -> `total_shares`